core/src/test/java/io/github/tcdl/ResponderTest.java

```diff
@@ -0,0 +1,103 @@
+package io.github.tcdl;
+
+import io.github.tcdl.config.MsbConfigurations;
+import io.github.tcdl.config.MsbMessageOptions;
+import io.github.tcdl.messages.Acknowledge;
+import io.github.tcdl.messages.Message;
+import io.github.tcdl.messages.MessageFactory;
+import io.github.tcdl.messages.payload.Payload;
+import io.github.tcdl.support.TestUtils;
+import io.github.tcdl.support.Utils;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+
+import static junit.framework.TestCase.assertEquals;
+import static org.mockito.Matchers.anyString;
+import static org.mockito.Mockito.*;
+
+
+/**
+ * Created by anstr on 5/26/2015.
+ */
+public class ResponderTest {
+    private MsbMessageOptions config;
+    MsbConfigurations msbConf;
+    private static final String TOPIC = "test:responder";
+
+    private ChannelManager mockChannelManager;
+    private Producer mockProducer;
+    private Acknowledge.AcknowledgeBuilder acknowledgeBuilder;
+
+    private Payload payload;
+    private Message originalMessage;
+    private Responder responder;
+
+    @Before
+    public void setUp() {
+        config = TestUtils.createSimpleConfig();
+
+        msbConf = TestUtils.createMsbConfigurations();
+
+        MessageFactory messageFactory = new MessageFactory(msbConf.getServiceDetails());
+        MessageFactory spyMessageFactory = spy(messageFactory);
+
+        MsbContext msbContext = TestUtils.createSimpleMsbContext();
+        MsbContext msbContextSpy = spy(msbContext);
```

Comment:
Why do you use 2 instances of MsbContext? MsbContext is almost a pojo. Why can't you inject mockChannelManager directly?